modules/javafx.controls/src/main/java/javafx/scene/control/Skinnable.java

```diff
@@ -36,29 +36,26 @@
  */
 public interface Skinnable {
     /**
-     * Skin is responsible for rendering this {@code Control}. From the
-     * perspective of the {@code Control}, the {@code Skin} is a black box.
-     * It listens and responds to changes in state in a {@code Control}.
+     * The Skin is responsible for rendering this {@code Skinnable}. From the
+     * perspective of the {@code Skinnable}, the {@code Skin} is a black box.
+     * It listens and responds to changes in state in a {@code Skinnable}.
      * <p>
-     * There is a one-to-one relationship between a {@code Control} and its
+     * There is typically a one-to-one relationship between a {@code Skinnable} and its
```

Comment:
The word "typically" doesn't describe when this applies and when not. From the previous comments, I get a better understanding (PopupControl), but developers reading the JavaDoc won't see that.

Reply:
Good point. We have a tension between "pure OO" and giving clear advice. In the case of `Skin`, there are only two direct subclasses, so we could call that out here and say that `Control` enforces a 1-to-1 relationship, but `PopupControl` does not. Alternatively, we could change this to say that "Some implementations of Skinnable define a 1-to-1 relationship...". That would key developers to look for that in the the documentation of the implementing class.

Which way do you think is best?

Reply:
I like "Some implementations of Skinnable define a 1-to-1 relationship..."...

Reply:
I re-phrased the javadoc, please see if it's clearer.